editor/import_defaults_editor.cpp

```diff
@@ -0,0 +1,217 @@
+		}
+
+		for (List<ResourceImporter::ImportOption>::Element *E = options.front(); E; E = E->next()) {
+			settings->properties.push_back(E->get().option);
+			if (d.has(E->get().option.name)) {
+				settings->values[E->get().option.name] = d[E->get().option.name];
+			} else {
+				settings->values[E->get().option.name] = E->get().default_value;
+			}
+			settings->default_values[E->get().option.name] = E->get().default_value;
+		}
+
+		save_defaults->set_disabled(false);
+		reset_defaults->set_disabled(false);
+
+	} else {
+		save_defaults->set_disabled(true);
+		reset_defaults->set_disabled(true);
+	}
+
+	settings->_change_notify();
+
+	inspector->edit(settings);
+}
+
+void ImportDefaultsEditor::_importer_selected(int p_index) {
+	_update_importer();
+}
+
+void ImportDefaultsEditor::clear() {
+	String last_selected;
+	if (importers->get_selected() > 0) {
+		last_selected = importers->get_item_text(importers->get_selected());
+	}
+
+	importers->clear();
+
+	importers->add_item("<" + TTR("Select Importer") + ">");
+	importers->set_item_disabled(0, true);
+
+	List<Ref<ResourceImporter> > importer_list;
+	ResourceFormatImporter::get_singleton()->get_importers(&importer_list);
+	Vector<String> names;
+	for (List<Ref<ResourceImporter> >::Element *E = importer_list.front(); E; E = E->next()) {
+		String vn = E->get()->get_visible_name();
+		names.push_back(vn);
+	}
+	names.sort();
+
+	for (int i = 0; i < names.size(); i++) {
+		importers->add_item(names[i]);
+
+		if (names[i] == last_selected) {
+			importers->select(i + 1);
+		}
+	}
+}
+
+void ImportDefaultsEditor::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("_reset"), &ImportDefaultsEditor::_reset);
+	ClassDB::bind_method(D_METHOD("_save"), &ImportDefaultsEditor::_save);
+	ClassDB::bind_method(D_METHOD("_importer_selected"), &ImportDefaultsEditor::_importer_selected);
+
+	ADD_SIGNAL(MethodInfo("project_settings_changed"));
+}
+
+ImportDefaultsEditor::ImportDefaultsEditor() {
+	HBoxContainer *hb = memnew(HBoxContainer);
+	hb->add_child(memnew(Label(TTR("Importer:"))));
+	importers = memnew(OptionButton);
+	hb->add_child(importers);
+	hb->add_spacer();
+	importers->connect("item_selected", this, "_importer_selected");
+	reset_defaults = memnew(Button);
+	reset_defaults->set_text(TTR("Reset to Defaults"));
+	reset_defaults->set_disabled(true);
+	reset_defaults->connect("pressed", this, "_reset");
+	hb->add_child(reset_defaults);
+	add_child(hb);
+	inspector = memnew(EditorInspector);
+	add_child(inspector);
+	inspector->set_v_size_flags(SIZE_EXPAND_FILL);
+	CenterContainer *cc = memnew(CenterContainer);
+	save_defaults = memnew(Button);
+	save_defaults->set_text(TTR("Save"));
+	save_defaults->connect("pressed", this, "_save");
+	cc->add_child(save_defaults);
+	add_child(cc);
+
+	settings = memnew(ImportDefaultsEditorSettings);
+}
+
+ImportDefaultsEditor::~ImportDefaultsEditor() {
+	if (inspector) {
+		inspector->edit(NULL);
```

Comment:
CC @qarmin

Sanitizer alerts about heap-use-after-free here, the problem is inconclusive. Advice?

Reply:
Probably moving this from destructor to notification predelete should fix use after free(I didn't tested it)
```
void ImportDefaultsEditor::_notification(int p_what) {
	if (p_what == NOTIFICATION_PREDELETE) {
		if (inspector) {
			inspector->edit(nullptr);
		}	
	}
}
```

Reply:
Thanks, problem solved!